mkwindows.sh

```diff
@@ -0,0 +1,228 @@
+    # - else, append ' -lc' to the config.
+    # I'm not sure why this is a good idea, but '-lc' is definitely wrong here.
+    sed -i -re 's%lt_cv_archive_cmds_need_lc=yes%lt_cv_archive_cmds_need_lc=no%' configure
+    #
+    # Disable linking against PNG, JPEG, TIFF, GIF, WIC,
+    # as those would either need cross-compilation, too, or some other magic.
+    ./configure -q --host x86_64-linux-gnu --target i686-w64-mingw32 \
+        --disable-dependency-tracking --prefix="${WEBP_DIR_FULL}/" \
+        --disable-static --enable-shared \
+        --enable-swap-16bit-csp --enable-experimental \
+        --disable-libwebpmux --disable-libwebpdemux \
+        --disable-libwebpdecoder --disable-libwebpextras \
+        --disable-png --disable-jpeg --disable-tiff --disable-gif --disable-wic \
+        CC=i686-w64-mingw32-gcc
+    # The warning
+    # "configure: WARNING: using cross tools not prefixed with host triplet"
+    # stems from the fact that 'mt' (magnetic tape control) is not available
+    # for i686-w64-mingw32, so the x86_64 version is used.  We can ignore that.
+    #
+    # Try to avoid too extreme autotools overhead:
+    touch Makefile.in
+    #
+    # Finally.
+    make --quiet -j4 install
+    #
+    # Fix bad naming by rebuilding that one file.  With blackjack and hookers!
+    rm -rf ../webp/lib
+    mkdir ../webp/lib
+    i686-w64-mingw32-gcc -shared -fPIC -DPIC  -Wl,--whole-archive src/dec/.libs/libwebpdecode.a src/dsp/.libs/libwebpdsp.a src/enc/.libs/libwebpencode.a src/utils/.libs/libwebputils.a -Wl,--no-whole-archive -lm -O2 -pthread -o ../webp/lib/libwebp.dll
+cd ../..
+if ! [ -r ${WEBP_DIR}/include/webp/decode.h -a -r ${WEBP_DIR}/lib/libwebp.a ] ; then
+    # I expect that cross-compiling webp is going to be very fragile,
+    # so print a nice error in case this happens.
+    echo "Cross-compilation apparently failed:"
+    echo "Some files in ${WEBP_DIR} are missing."
+    echo "Please submit a bugreport."
+fi
+
+
+# === Build auto-gen files ===
+
+# Create and backup all auto/ files
+echo "===== 2: Create auto/ files"
+make tgl/Makefile
+make -C tgl -j4 auto/auto-autocomplete.c auto/auto.c auto/auto-fetch.c auto/auto-fetch-ds.c auto/auto-free-ds.c
+make -C tgl -j4 auto/auto-print-ds.c auto/auto-skip.c auto/auto-store.c auto/auto-store-ds.c auto/auto-types.c
+make bin
+tar cf bin/tgl-auto-code.tar tgl/auto/
+
+
+# === Cross-compilation itself ===
+
+# Clean tgl only (do not touch bin/ with the auto-gen files!)
+echo "===== 3: Cleanup tgl"
+make -C tgl clean
+
+# Reconfigure
+echo "===== 4: Configure for cross-compilation"
+# Sadly, the library paths must be resolved *now* already.
+# AND, they must be absolute.  Sigh.
+export WIN32_GTK_DEV_DIR=`realpath win32/gtk_2_0-2.16`
+export WIN32_PIDGIN_DIR=`realpath win32/pidgin-2.12.0`
+export WIN32_WEBP_DIR="${WEBP_DIR_FULL}"
+./configure -q --host x86_64-linux-gnu --target i686-w64-mingw32 \
+    --includedir="/usr/share/mingw-w64/include" \
```

Comment:
The first path?  No, since it's a build directory, but `configure` really dislikes relative paths.

The second one, given to `--includedir`?  Yeah, probably, I'll look into how well that works.  Thanks!